config/console/console_init.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: console
+spec:
+  selector:
+    matchLabels:
+      app: mcg-ms-console
+  template:
+    metadata:
+      labels:
+        app: mcg-ms-console
+    spec:
+      containers:
+        - name: mcg-ms-console
+          image: mcg-ms-console:latest
+          resources:
+            limits:
+              cpu: "100m"
+              memory: "512Mi"
+          ports:
+            - containerPort: 9002
+              protocol: TCP
+          volumeMounts:
+            - name: addon-mcg-osd-parameters
+              mountPath: /var/serving-cert
+              readOnly: true
+      tolerations:
+      - effect: NoSchedule
+        key: node.ocs.openshift.io/storage
+        operator: Equal
+        value: "true"
+      volumes:
+        - name: addon-mcg-osd-parameters
+          secret:
+            secretName: addon-mcg-osd-parameters
```

Comment:
I think the reason why the TLS artifacts are not being created is because we didn't create a `secret` with this name anywhere in the first place. If you look [here](https://dev.to/joshduffney/kubernetes-using-configmap-subpaths-to-mount-files-3a1i), they use a similar approach (`configmap` instead of `secret`, but it does not matter since the created files will have the name as the `configmap/secret.Data`'s keys and the contents as the respective values for those keys).

Basically, we need to make sure that the dummy `secret` is present with the required data before the execution reaches this point.

Reply:
Yes, we need to create the secret externally when we run it manually. sample secret:

```
kind: Secret
apiVersion: v1
metadata:
  name: addon-mcg-osd-parameters
  namespace: fed
data:
  tls.crt: "base64 encoded cert"
  tls.key: "base64 encoded key"
type: kubernetes.io/tls
```